test/with-timers.test.js

```diff
@@ -0,0 +1,43 @@
+import t from 'tap'
+import Fastify from 'fastify'
+import FakeTimers from '@sinonjs/fake-timers'
+
+import slowDownPlugin from '../index.js'
+import { convertToMs, slowDownAPI } from '../lib/helpers.js'
+import { HEADERS, DEFAULT_OPTIONS } from '../lib/constants.js'
+
+t.test('Check the response time with 2 enpoints', async t => {
+  const clock = FakeTimers.install()
+  const fastify = Fastify()
+  await fastify.register(slowDownPlugin)
+  t.teardown(() => {
+    clock.uninstall()
+    fastify.close()
+  })
+
+  fastify.get('/', async () => 'Hello fastify-slow-down!')
+  fastify.get('/fastify', () => 'Welcome fastify-slow-down!')
+
+  await slowDownAPI(() => fastify.inject('/'))
+
+  const delayMs = convertToMs(DEFAULT_OPTIONS.delay)
+
+  clock.tick(1)
+  let response = await fastify.inject('/')
+  t.equal(response.statusCode, 200)
+  t.equal(response.headers[HEADERS.delay], delayMs)
+
+  response = await fastify.inject('/fastify')
+  t.equal(response.statusCode, 200)
+  t.equal(response.headers[HEADERS.delay], 2 * delayMs)
+
+  clock.tick(1000)
+  response = await fastify.inject('/fastify')
+  t.equal(response.statusCode, 200)
+  t.equal(response.headers[HEADERS.delay], 3 * delayMs)
+
+  clock.tick(convertToMs(DEFAULT_OPTIONS.timeWindow) + 1000)
+  response = await fastify.inject('/')
+  t.equal(response.statusCode, 200)
+  t.equal(response.headers[HEADERS.remaining], DEFAULT_OPTIONS.delayAfter - 1)
```

Comment:
this assertion is hard to read, you're testing different things in the same test in a very chaotic way, which makes it hard to understand what you're testing